src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx

```diff
@@ -253,8 +253,29 @@ export function GeneralEntityRoot<T extends BaseEntity>({
     }
   }, [entityController, onEntitiesLoaded, entityType, pageCurrent, pageSize, sortField, sortOrder, isFiltered]);
 
+  const discardChanges = () => {
+    Modal.destroyAll();
+    fetchEntity(parseInt(id.toString(), 10));
+    notification.info({
+      message: t('GeneralEntityRoot.saveWarning', {
+        entity: TranslationUtil.getTranslationFromConfig(entityName, i18n)
+      })
+    });
+  };
+
+  const saveChanges = () => {
+    onSaveClick();
+    fetchEntity(parseInt(id.toString(), 10));
+    Modal.destroyAll();
+  };
+  const reviewChanges = () => {
+    Modal.destroyAll();
+    setFormIsDirty(true);
```

Comment:
I am wondering if you need to reset the id? I assume the entity is still intact, so something like:
```
setId(entity.id)
```

Reply:
ok, great